### PR TITLE
#3957 Add typing ID to specific pathogen test types

### LINKED ISSUE #3957: Add typing ID to specific pathogen test types [1]
* Add new field `typingId` (text) to pathogen tests
* Add a new pathogen test type called "DNA Microarray"
* Only show the new field when one of the following pathogen test types is selected: PCR, Sequencing, DNA Microarray

### PR DESCRIPTION
**Pull request by DavidBaldsiefen**
Closes #3957

### CODE REVIEW

File: sormas-app/app/src/main/res/layout/fragment_pathogen_test_read_layout.xml [hunk 1]
```diff
@@ -73,6 +73,19 @@
 
                 </LinearLayout>
 
+                <LinearLayout
+                        android:layout_width="match_parent"
+                        android:layout_height="wrap_content"
+                        android:orientation="horizontal">
+
+                    <de.symeda.sormas.app.component.controls.ControlTextReadField
```
Comment (MateStrysewske) [inline sormas-app/app/src/main/res/layout/fragment_pathogen_test_read_layout.xml:81]: Typing ID here should also only be visible for the three pathogen test types; also, you don't need a wrapping LinearLayout when you use the ControlSingleColumnStyle

**Review by MateStrysewske — CHANGES_REQUESTED** [review]

**Review by tkaefer — COMMENTED** [review]
LGTM, besides the remarks from @MateStrysewske

**Comment by MateStrysewske** [comment]
@DavidBaldsiefen Can you solve the merge conflicts again please? :)

**Comment by DavidBaldsiefen** [comment]
@MateStrysewske done

**Review by MateStrysewske — APPROVED** [review]

**Pull request merged**